frontend/apps/reader/modules/readersearch.lua

```diff
@@ -79,6 +80,12 @@ SRELL_ERROR_CODES[110] = _("No preceding expression in repetition.")
 SRELL_ERROR_CODES[111] = _("Expression too complex, some hits will not be shown.")
 SRELL_ERROR_CODES[666] = _("Expression may lead to an extremely long search time.")
 
+function ReaderSearch:registerKeyEvents()
+    if Device:hasKeyboard() then
+        self.key_events.ShowFulltextSearchInput = { { Device.input.group.AlphaNumeric } }
```

Comment:
I'm not really using any shorcut on the emulator - and was somehow ok that you removed any direct stroke (ie. <kbd>B</kbd> showing bookmarks) so one won't trigger any unwanted popup while he is reading and resting his fingers on the keyboard, and sometimes pressing a bit too much or just finger tapping out of excitement... until I realized that now, ANY alphabetical stroke will trigger the full text search.
Just pointing that out so people can say whether they feel this is ok.
No strong opinion, but I feel it is a bit invasive that any key will trigger full search, just to save one keystroke, ie <kbd>s</kbd>.
Also, people that got user to typing B, H, J, K, T won't get what they were used to.

Reply:
In actuality, only `B`, `T` and `H`are currently mapped to anything that does something.

Regarding the ANY triggers search, it makes the whole thing feel more polished, like there is a purpose to having all those keys, i was also inspired by stock firmware that does the same thing. Although they don't have a fragmented search... maybe we could get a universal search, please 🙏

Reply:
> Just pointing that out so people can say whether they feel this is ok.

No, I think that sounds very undesirable.

Reply:
Undesirable on kindle too? I think it's quite nice there. (I could be wrong of course)

Reply:
Especially there. Reducing 26 or more action buttons to merely one is surely worse if you don't have touch or mouse.

Reply:
Quite the contrary I'd wager. First we are not reducing 26 actions to one. The keyboard is realistically not used for anything besides perhaps for these three keys, "B", "T" and "H" which i think, should never have been shortcuts on their own (without modifiers, on kindle at least). Incidentally (and as stated before), it is the same behaviour stock firmware has (i.e alphanumeric keys call up the search function). 

The only thing that we'd need (should this were to make it) would be to not remember the last search (only on kindle with keyboards of course) so one can type and search without needing to delete anything

Reply:
The design is similar in principle to for example:

https://man.cx/mupdf(1)
https://man.cx/zathura(1)
https://github.com/johnfactotum/foliate/blob/479e0d5f3e2e7e5e6edffce3487eac86ce7fdfea/src/book-viewer.js#L677-L706

> The keyboard is realistically not used for anything besides perhaps for these three keys, "B", "T" and "H" which i think, should never have been shortcuts on their own (without modifiers, on kindle at least). Incidentally (and as stated before), it is the same behaviour stock firmware has (i.e alphanumeric keys call up the search function).

Try to move past "Kindle does it". Why do you think search is so important that one shouldn't be allowed easy access to bookmarks, rotation, zoom mode, or any other action in the gestures plugin?

Reply:
> The design is similar in principle to for example:

I understand the convenience of single-button shortcuts in applications like Photoshop and Lightroom. However, I find them less suitable for Kindle devices. Accidental triggers can lead to unexpected outcomes, such as unintentionally rotating the screen by 45 degrees (a potential catastrophic outcome on older hardware ;) ). Furthermore, having numerous keys that serve no purpose feels like a missed opportunity—a bit like building roads that lead nowhere.

While I don't believe the search function is 26 times better than others, I appreciate a well-rounded system. Rather than having idle keys, let's make the most of our keyboard. On a related note, I debated implementing similar changes on FM, but I'd miss the shortcuts for opening books, which I've grown quite fond of... but maybe (for the greater good) it should also start a search query

Reply:
I have this disabled in my Firefox, but it feels it's what you want: https://kb.mozillazine.org/Accessibility.typeaheadfind
For it to work, you would need the first letter you hit to also be inserted in the searchbox - which I don't think is happening (and I don't want to think about how you could have that - it would possibly add kludge.).

> Rather than having idle keys, let's make the most of our keyboard.

That makes the keyboard behave as a single same key :) quite well not rounded :)
I still don't know about all the kind of kindles (keyboard + dpad, or only one of them), but remembering all the non-obvious shortcuts you added using combinations of Shift/Sym/..., that may be you have learned, but I'm not sure other users will, it feels that just adding all our various features as keys (ie. P=PageBrowser, B=Bookmarks, M=BookMap...)  would be a more well rounded system. (There's just one bad thing with that is that it's intuitive only for English users :/)

Reply:
>  you would need the first letter you hit to also be inserted in the searchbox

it is happening actually. it works perfectly.

Reply:
I had to verify that, because nothing in the code looks like this should happen.
On the emulator, it works only the first time after launch - but no longer on the next attemps.
(Could be some state unset until first launch, or key press launching the popup, and key release inserting the char, somehow prevented after the first?)

Reply:
> it works only the first time after launch - but no longer on the next attemps.

not quite sure what that means... are you saying that it automagically works on kindle but not elsewhere? that has to be some signal from the gods. btw check out my edit in the previous comment (above)

Reply:
> However, I find them less suitable for Kindle devices.

I feel like a broken record but *this was explicitly implemented **for** Kindle devices. This is not some weird bloody unintentional side effect.

Reply:
> are you saying that it automagically works on kindle but not elsewhere? that has to be some signal from the gods. btw check out my edit in the previous comment (above)

I'm just saying it does not work on 2nd++ attemps on my emulator - I'm trusting you (with difficulties :)) that it works on Kindle. I can't understand why this would work.

> the only thing that would need to change is this 

I guess you're talking about starting from an empty search field when invokied from a key (whether we end up assiging all keys or only <kbd>s</kbd>).
I guess you could pass an arg empty_input=true with the event so if true, ilne 217 sets it to "".

Reply:
On Kindle keypresses are handled fully by us, while on SDL we have a distinction between "input" and "keypress". That's probably what leads to the discrepancy.

Reply:
> I feel like a broken record but this was explicitly implemented for Kindle devices. This is not some weird bloody unintentional side effect.

Yes, I understand that, that was many moons ago. The question now is: do you want that _status quo_ support to continue (only a handful shortcuts) or a more rounded up feature AND shortcuts which are not disappearing?

I am trying this but it does not work...
```lua
local empty_input = true
self.key_events.ShowFulltextSearchInput = { { Device.input.group.AlphaNumeric }, args = { empty_input } }
```

Reply:
Looking at some examples using args in readerpaging, it may need to be:
```lua
self.key_events.BlahBlahShowFulltextSearchInput = {
   { Device.input.group.AlphaNumeric },
   event = "ShowFulltextSearchInput",
   args = true,
}

[...]
function ReaderSearch:onShowFulltextSearchInput(empty_input)
[...]
    input = empty_input and "" or self.last_search_text or self.ui.doc_settings:readSetting("fulltext_search_last_search_text"),
```

Reply:
I also moved the `is_enter_default = true,` from right arrow to `All`, makes more sense...

Reply:
> I also moved the is_enter_default = true, from right arrow to All, makes more sense...

I disagree (I guess your `...` meant you were expecting that :))
1) it is a lot more expensive, so some delay
2) The `>` is the one on the right, which is usually what the default is. `All` is in the middle, not obvious that this will be the default.
3) People may be used to the current behaviour.

Reply:
> The question now is: do you want that _status quo_ support to continue (only a handful shortcuts) or a more rounded up feature AND shortcuts which are not disappearing?

Several contributors have already stated they do not appreciate false dilemmas. I don't either.

Reply:
> (I guess your ... meant you were expecting that :))

Possibly.

I have never found that version of search useful, I personally prefer the full list, therefore I never use it. It's a shame since reaching the "All" requires quite a few presses instead of simply pressing "Enter"